Add swagger documentation

Purhan opened the pull request:
# Description
Added swagger API docs using drf-yasg. Preview below:
![image](https://user-images.githubusercontent.com/10897423/109160443-cee9d800-779b-11eb-8f13-76d7138c4fa6.png)


## Related issues
Please add related issues.
#323

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue).
- [x] New feature (non-breaking change which adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected).

# Checklist

- [x] The pull request is for the branch develop
- [ ] If I added a new analyzer, I updated the file [Usage](https://github.com/intelowlproject/IntelOwl/blob/master/docs/source/Usage.md). If the analyzer provides additional optional configuration, I added the available options here: [Advanced-Usage](./Advanced-Usage.md)
- [ ] If I added external libraries/packages that use restrictive licenses, please add them in the [ReadMe - Legal Notice](https://github.com/certego/IntelOwl/blob/master/README.md) section
- [ ] I added new secrets in the files [env_file_app_template](https://github.com/intelowlproject/IntelOwl/blob/master/env_file_app_template), [env_file_app_ci](https://github.com/certego/IntelOwl/blob/master/env_file_app_travis) and in the docs: [Installation](./Installation.md)
- [x] I have added tests in the [Tests](https://github.com/intelowlproject/IntelOwl/blob/master/tests) folder. 
- [x] The tests gave 0 errors.
- [x] `Black` gave 0 errors.
- [x] `Flake` gave 0 errors.
- [x] I squashed the commits into a single one.
  
### please follow these rules
- If your changes decrease the overall tests coverage (you will know after the Codecov CI job is done), you should add the required tests to fix the problem
- Everytime you make changes to the PR and you think the work is done, you should explicitly ask for a review

mlodic reviewed (commented):
thank you for the PR!

It is possible to set up the authorization via API token? I ask this because ATM it is only possible to test the APIs from the swagger docs by only using the Basic Auth that is not supported by IntelOwl

![image](https://user-images.githubusercontent.com/30625432/109164870-33e10600-777b-11eb-9e7f-d065f412574e.png)

Also I think we should rewrite the docstrings of the APIs because swagger cannot parse that data correctly. Ex:
![image](https://user-images.githubusercontent.com/30625432/109165226-9afeba80-777b-11eb-9302-d94cf8b60b4c.png)

Purhan commented on intel_owl/settings.py:
```diff
@@ -12,7 +12,7 @@
 
 # SECURITY WARNING: don't run with debug turned on in production!
-DEBUG = os.environ.get("DEBUG", False) == "True"
+DEBUG = True
```
omg how did I miss this :laughing:, will revert it soon.

Purhan commented:
> Is it possible to set up the authorization via API token? I ask this because ATM it is possible to test the APIs from the swagger docs only by the Basic Auth that is not supported by IntelOwl

Yes, we can use any permission class from the Rest-API framework.

Purhan commented:
I have reverted the unrelated changes, added an authentication class so that it can't be accessed without the authentication token. Now what's left is changing the docstrings. Will update soon.

mlodic commented:
Am I me only or you get "401 Unauthorized" too when going to `/api/docs`?

Eshaan7 commented:
Hi. Thankyou for your contribution. However, there are a few caveats in the way you implemented this is very different from our use case.

If you see #332 and  [this comment](https://github.com/intelowlproject/IntelOwl/pull/332#issuecomment-770382482)., someone already did this and we had to close that PR too for same reason.

> Use [this](https://drf-yasg.readthedocs.io/en/stable/rendering.html#management-command) to create the spec file, and then use the `sphinx-redoc` extension to generate redoc along with rest of the docs. Then link both docs together with [something like this](https://github.com/sphinx-contrib/redoc/issues/15). There should be a GH workflow setup for this.

Purhan commented:
I see, thank you for reviewing. I'm sorry but I can't work on this anymore due to personal engagements. Closing this for now. Thank you for your time @Eshaan7 @mlodic :+1:

Eshaan7 commented:
> Also I think we should rewrite the docstrings of the APIs because swagger cannot parse that data correctly.

Yes, it's our mistake for not following a proper syntax while writing doc strings. I like the [napoleon syntax](https://sphinxcontrib-napoleon.readthedocs.io/en/latest/).

Need to use proper syntax as given [here](https://drf-yasg.readthedocs.io/en/1.0.4/custom_spec.html).
> descriptions for Operations, Parameters and Schemas are picked up from docstrings and help_text attributes in the same manner as the default DRF SchemaGenerator
